src/main/java/io/github/cottonmc/cotton/gui/widget/WButton.java

```diff
@@ -14,63 +14,62 @@
 	protected int darkmodeColor = WLabel.DEFAULT_TEXT_COLOR;
 	private boolean enabled = true;
 	protected Alignment alignment = Alignment.CENTER;
-	
+
 	private Runnable onClick;
-	
+
 	public WButton() {
-		
+
 	}
-	
+
 	public WButton(Text text) {
 		this.label = text;
 	}
-	
+
 	@Override
 	public boolean canResize() {
 		return true;
 	}
-	
+
 	@Override
 	public void paintBackground(int x, int y, int mouseX, int mouseY) {
 		boolean hovered = (mouseX>=0 && mouseY>=0 && mouseX<getWidth() && mouseY<getHeight());
 		int state = 1; //1=regular. 2=hovered. 0=disabled.
 		if (!enabled) state = 0;
 		else if (hovered) state = 2;
-		
+
 		float px = 1/256f;
 		float buttonLeft = 0 * px;
 		float buttonTop = (46 + (state*20)) * px;
 		int halfWidth = getWidth()/2;
 		if (halfWidth>198) halfWidth=198;
 		float buttonWidth = halfWidth*px;
 		float buttonHeight = 20*px;
-		
+
 		float buttonEndLeft = (200-(getWidth()/2)) * px;
-		
-		ScreenDrawing.texturedRect(x, y, getWidth()/2, 20, AbstractButtonWidget.WIDGETS_LOCATION, buttonLeft, buttonTop, buttonLeft+buttonWidth, buttonTop+buttonHeight, 0xFFFFFFFF);
-		ScreenDrawing.texturedRect(x+(getWidth()/2), y, getWidth()/2, 20, AbstractButtonWidget.WIDGETS_LOCATION, buttonEndLeft, buttonTop, 200*px, buttonTop+buttonHeight, 0xFFFFFFFF);
-		
+
+		ScreenDrawing.texturedRect(x, y, getWidth()/2, getHeight(), AbstractButtonWidget.WIDGETS_LOCATION, buttonLeft, buttonTop, buttonLeft+buttonWidth, buttonTop+buttonHeight, 0xFFFFFFFF);
+		ScreenDrawing.texturedRect(x+(getWidth()/2), y, getWidth()/2, getHeight(), AbstractButtonWidget.WIDGETS_LOCATION, buttonEndLeft, buttonTop, 200*px, buttonTop+buttonHeight, 0xFFFFFFFF);
+
 		if (label!=null) {
 			int color = 0xE0E0E0;
 			if (!enabled) {
 				color = 0xA0A0A0;
 			} /*else if (hovered) {
 				color = 0xFFFFA0;
 			}*/
-			
 			ScreenDrawing.drawStringWithShadow(label.asFormattedString(), alignment, x, y + ((20 - 8) / 2), width, color); //LibGuiClient.config.darkMode ? darkmodeColor : color);
 		}
 	}
-	
+
 	@Override
 	public void setSize(int x, int y) {
-		super.setSize(x, 20);
+		super.setSize(x, y);
```

Comment:
No. There's a reason why the height is hardcoded to 20, and the texture with be broken with different heights.

Reply:
Seemed to work fine with a smaller size for me. I'd show you but, i ended up not using a button anyway so I'll just revert it